packages/back-end/src/controllers/stripe.ts

```diff
@@ -261,3 +270,118 @@ export async function postWebhook(req: Request, res: Response) {
+    const setupIntent = await stripe.setupIntents.create({
+      customer,
+      payment_method_configuration: STRIPE_PAYMENT_METHOD_CONFIG_ID,
+      automatic_payment_methods: {
+        enabled: true,
+      },
+    });
+    return res.status(200).json({ clientSecret: setupIntent.client_secret });
+  } catch (e) {
+    throw new Error(e.message);
+  }
+}
+
+export async function postStripeCustomerDefaultCard(
+  req: AuthRequest<{ paymentMethodId: string; subscriptionId: string }>,
+  res: Response
+) {
+  const { paymentMethodId, subscriptionId } = req.body;
+
+  try {
+    const customerId = await getStripeCustomerId(subscriptionId);
+    await stripe.customers.update(customerId, {
+      invoice_settings: { default_payment_method: paymentMethodId },
+    });
+  } catch (e) {
+    throw new Error(e.message);
+  }
+  res.status(200).json({
+    status: 200,
+  });
+}
+
+export async function listPaymentMethods(
+  req: AuthRequest<null, { subscriptionId: string }>,
+  res: Response
+) {
+  try {
+    // Fetch the customer to get default_payment_method
+    const customerId = await getStripeCustomerId(req.params.subscriptionId);
+    const customer = await stripe.customers.retrieve(customerId);
+
+    if (customer.deleted) {
+      return res.status(200).json({ status: 200, cards: [] });
+    }
+    let defaultPaymentMethod: undefined | string = undefined;
+    const paymentMethod = customer.invoice_settings.default_payment_method;
+
+    if (paymentMethod) {
+      if (typeof paymentMethod === "string") {
+        defaultPaymentMethod = paymentMethod;
+      } else {
+        defaultPaymentMethod = paymentMethod.id;
+      }
+    }
+
+    // Fetch all card payment methods for customer
+    const { data: paymentMethods } = await stripe.customers.listPaymentMethods(
+      customerId,
+      {
+        type: "card",
+      }
+    );
+
+    if (!paymentMethods) {
+      return res.status(200).json({ status: 200, cards: [] });
+    }
+
+    const cards: Card[] = paymentMethods
+      .map((method, i) => {
+        const card = method.card;
+        if (!card) return undefined;
+
+        return {
+          id: method.id,
+          last4: card.last4,
+          brand: card.brand,
+          expMonth: card.exp_month,
+          expYear: card.exp_year,
+          // if no explicit defaultPaymentMethodId, Orb uses the first card, so matching that logic
+          isDefault: defaultPaymentMethod
+            ? method.id === defaultPaymentMethod
+            : !!(i === 0),
+        };
+      })
+      .filter((card): card is Card => Boolean(card));
+
+    return res.status(200).json({ status: 200, cards });
+  } catch (e) {
+    throw new Error(e.message);
+  }
+}
+
+export async function deletePaymentMethod(
+  req: AuthRequest<{ paymentMethodId: string }>,
+  res: Response
+) {
+  const { paymentMethodId } = req.body;
+
+  try {
+    await stripe.paymentMethods.detach(paymentMethodId);
```

Comment:
We should do an extra sanity check here to make sure the payment method is owned by the customer.  Otherwise, any user could delete any other user's credit card just by guessing the paymentMethodId.